go-controller/pkg/factory/handler.go

```diff
@@ -160,28 +161,35 @@ func (i *informer) processEvents(events chan *event, stopChan <-chan struct{}) {
 	}
 }
 
-func getQueueNum(oType reflect.Type, obj interface{}) uint32 {
+func (i *informer) getQueueNum(oType reflect.Type, obj interface{}, numEventQueues uint32, del bool) uint32 {
+
 	meta, err := getObjectMeta(oType, obj)
 	if err != nil {
 		klog.Errorf("Object has no meta: %v", err)
 		return 0
 	}
 
-	// Distribute the object to an event queue based on a hash of its
-	// namespaced name, so that all events for a given object are
-	// serialized in one queue.
-	h := fnv.New32()
-	if meta.Namespace != "" {
-		_, _ = h.Write([]byte(meta.Namespace))
-		_, _ = h.Write([]byte("/"))
+	if del {
```

Comment:
It looks like you're picking a queue based on the UID of the object, rather than the namespace+name. I wouldn't do that.

It means that we could lose ordering for a delete+add of the same object. If the handler processing the delete happens to be slower, the add might happen first.